### DISCUSSION TITLE
Document zod type coercion

### DISCUSSION THREAD

**Discussion by fiws** [edmundhung/conform#337] [Ideas]
I found myself reading the source code a couple times now, because i couldn't find any documentation for the automatic zod type coercion.

Number 1 thing i constantly have to lookup is how booleans work. Is it "ON", "yes", "1", "On", "true"?

I know that i don't need to know the internals if i used the conform helper for everything. I still would like to have it documented. Sometimes it's just easier to use plain form elements directly. It also is required to know if you plan to do some API testing without using the form directly.

**Comment by aaronadamsCA** [comment]
I too was surprised to discover that built-in transforms already convert form data values like empty files and empty strings to `undefined`. I had already written my own transforms to convert these to `null`, which is what I need, so I couldn't figure out why `undefined` was still showing up in the types.

Noting it here in case it helps others.
